src/sentry/static/sentry/app/views/organizationSettings.jsx

```diff
@@ -0,0 +1,356 @@
+import React from 'react';
+
+import ApiMixin from '../mixins/apiMixin';
+import {
+  BooleanField,
+  FormState,
+  Select2Field,
+  TextField,
+  TextareaField
+} from '../components/forms';
+import IndicatorStore from '../stores/indicatorStore';
+import LoadingIndicator from '../components/loadingIndicator';
+import OrganizationHomeContainer from '../components/organizations/homeContainer';
+import {t} from '../locale';
+
+
+const OrganizationSettingsForm = React.createClass({
+  propTypes: {
+    orgId: React.PropTypes.string.isRequired,
+    access: React.PropTypes.object.isRequired,
+    initialData: React.PropTypes.object.isRequired,
+    onSave: React.PropTypes.func.isRequired
+  },
+
+  mixins: [ApiMixin],
+
+  getInitialState() {
+    return {
+      formData: this.buildFormData(this.props.initialData),
+      errors: {},
+      hasChanges: false,
+    };
+  },
+
+  buildFormData(data) {
+    let result = {
+      name: data.name,
+      slug: data.slug,
+      openMembership: data.openMembership,
+      allowSharedIssues: data.allowSharedIssues,
+      isEarlyAdopter: data.isEarlyAdopter,
+      enhancedPrivacy: data.enhancedPrivacy,
+      dataScrubber: data.dataScrubber,
+      dataScrubberDefaults: data.dataScrubberDefaults,
+      scrubIPAddresses: data.scrubIPAddresses,
+      safeFields: data.safeFields.join('\n'),
+      sensitiveFields: data.sensitiveFields.join('\n'),
+    };
+    if (this.props.access.has('org:delete')) {
+      result.defaultRole = data.defaultRole;
+    }
+    return result;
+  },
+
+  onFieldChange(name, value) {
+    let formData = this.state.formData;
+    formData[name] = value;
+    this.setState({
+      hasChanges: true,
+      formData: formData,
```

Comment:
Note this will not necessarily trigger a state change, since the `formData` reference did not change.

I'd force a new reference by rebuilding `formData`:

```javascript
this.setState({
  hasChanges: true,
  formData: {...formData} // same as Object.assign({}, formData);
});